bin/generate-receipts.js

```diff
@@ -0,0 +1,214 @@
+        return await BillingProperty.create(
+            this.context,
+            {
+                dv: DV,
+                sender: SENDER,
+                context: { connect: { id: this.billingContextId } },
+                address: buildFakeAddressMeta().value,
+                importId: faker.datatype.uuid(),
+                globalId: faker.datatype.uuid(),
+                meta: {},
+                raw: {},
+            }
+        )
+    }
+	
+    async generateBillingAccountsForProperty (property) {
+        const { id } = property
+        const accountsToGenerate = Math.floor(Math.random() * (this.accountsPerProperty.max - this.accountsPerProperty.min + 1)) + this.accountsPerProperty.min
+        console.info(`[INFO] ${accountsToGenerate} would be generated for ${property.address}`)
+        
+        const billingAccounts = []
+        for (let i = 0; i < accountsToGenerate; ++i) {
+            const billingAccount = await BillingAccount.create(
+                this.context,
+                {
+                    dv: DV,
+                    sender: SENDER,
+                    context: { connect: { id: this.billingContextId } },
+                    property: { connect: { id: id } },
+                    importId: faker.datatype.uuid(),
+                    globalId: faker.helpers.replaceSymbols('##??########'),
+                    number: faker.helpers.replaceSymbols('############'),
+                    unitName: (i + 1).toString(),
+                    meta: BASE_JSON,
+                    raw: BASE_JSON,
+                }
+            )
+            billingAccounts.push(billingAccount)
+        }
+
+        return billingAccounts
+    }
+
+    async generateReceipts () {
+
+        // Gets somewhat like normal distribution using central limit theorem
+        function _gaussianRand () {
+            let rand = 0
+            for (let i = 0; i < 6; i += 1) {
+                rand += Math.random()
+            }
+            return rand / 6
+        }
+        
+        console.info('[INFO] Generating receipts...')
+        if (!this.billingAccounts) {
+            throw new Error('Cant generate receipts! No billing accounts. Please check that this.prepareModels() is ran before this.generateReceipts()')
+        }
+
+        const toBeGenerated = this.billingAccounts.length
+
+        // Right now we support only one payment organization
+        // The case when there are multiple payment organizations is possible
+        // todo(toplenboren) create multiple payment receipts for different orgs for same account
+        const PAYMENT_ORGANIZATION = {
+            tin: faker.datatype.number().toString(),
+            iec: faker.datatype.number().toString(),
+            bic: faker.datatype.number().toString(),
+            bankAccount: faker.datatype.number().toString(),
+        }
+
+        let currentProgress = 0 // in percentage
+
+        setInterval(() => {
+            process.stdout.write(`\r${currentProgress}%`)
```

Comment:
why not console.log if using console.info above?

Reply:
This lad allows us to make the contitious loader